tests/kernel/spinlock/src/spinlock_error_case.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <ztest.h>
+#include <spinlock.h>
+
+BUILD_ASSERT(CONFIG_MP_NUM_CPUS > 1);
+
+static struct k_spinlock lock;
+static struct k_spinlock mylock;
+static k_spinlock_key_t key;
+
+static ZTEST_DMEM volatile bool valid_assert;
+
+static inline void set_assert_valid(bool valid)
+{
+	valid_assert = valid;
+}
+
+static void action_after_assert_fail(void)
+{
+	k_spin_unlock(&lock, key);
+
+	ztest_test_pass();
+}
+
+#ifdef CONFIG_ASSERT_NO_FILE_INFO
+void assert_post_action(void)
+#else
+void assert_post_action(const char *file, unsigned int line)
+#endif
+{
+#ifndef CONFIG_ASSERT_NO_FILE_INFO
+	ARG_UNUSED(file);
+	ARG_UNUSED(line);
+#endif
+
+	printk("Caught assert failed\n");
```

Comment:
I guess that it should be "Caught an assert."

Reply:
Thanks for correcting this, got it fix!